pkgs/development/python-modules/scipy/default.nix

```diff
@@ -1,34 +1,41 @@
 {
   lib,
-  stdenv,
-  fetchFromGitHub,
-  fetchpatch,
   fetchurl,
   writeText,
-  xcbuild,
   python,
   buildPythonPackage,
+  fetchFromGitHub,
+  fetchpatch,
+  stdenv,
+
+  # build-system
   cython,
   gfortran,
   meson-python,
   nukeReferences,
-  pkg-config,
   pythran,
-  wheel,
+  pkg-config,
   setuptools,
-  hypothesis,
-  pytest7CheckHook,
-  pytest-xdist,
-  numpy,
-  pybind11,
-  pooch,
-  xsimd,
+  xcbuild,
+
+  # buildInputs
   # Upstream has support for using Darwin's Accelerate package. However this
   # requires a Darwin user to work on a nice way to do that via an override.
   # See:
   # https://github.com/scipy/scipy/blob/v1.14.0/scipy/meson.build#L194-L211
   blas,
   lapack,
+  pybind11,
+  pooch,
+  xsimd,
```

Comment:
What is the value in moving these here?

Reply:
I followed the order of apparition in the derivation and grouped everything by category.

Reply:
> I followed the order of apparition in the derivation and grouped everything by category.

I understand that this is the rational, but still they could have been left there as they are `buildInputs` - you could have just added a `# buildInputs` title.

Reply:
Yes sure, but as we started messing with the arguments, I thought legitimate to clean them exhaustively.

Reply:
> Yes sure, but as we started messing with the arguments, I thought legitimate to clean them exhaustively.

I'm sorry, I value much more small diffs then completely arbitrary orderings of inputs.

Reply:
> I'm sorry, I value much more small diffs then completely arbitrary orderings of inputs.

This is very legit.
I tend to do those changes when I have to touch the derivation in question because I feel like they do not deserve their own PRs.
If you would rather have them in a separate one, it would be perfectly fair.

Reply:
> > I'm sorry, I value much more small diffs then completely arbitrary orderings of inputs.
> 
> This is very legit. I tend to do those changes when I have to touch the derivation in question because I feel like they do not deserve their own PRs. If you would rather have them in a separate one, it would be perfectly fair.

It's OK, you got me convinced mostly.